CMakeLists.txt

```diff
@@ -79,9 +80,154 @@ else()
     add_subdirectory(external/lyra)
     add_subdirectory(external/rang)
 endif()
-add_subdirectory(thorin)
+
+add_library(libthorin)
+target_sources(libthorin
```

Comment:
This seems error-prone. Can't we just say to include all cpp and h files?

Reply:
This is sth I hate about cmake but all guides say that you really, really shouldn't use globbing and manually add all sources. See e.g. [here](https://stackoverflow.com/questions/1027247/is-it-better-to-specify-source-files-with-glob-or-each-file-individually-in-cmak).